Protocol.md

```diff
@@ -627,6 +631,13 @@ subprotocol strings, and:
   SHALL validate that the negotiated subprotocol is present in the
   *subprotocols* field.
 
+Furthermore, the server SHALL validate that the *ping_interval* field
+contains a non-negative integer. If the value is `0`, the server SHALL
+NOT send WebSocket *ping* messages to the client. Otherwise, the server
+SHOULD send a WebSocket *ping* message in the requested interval to the
+client. A timeout of 30 seconds for unanswered *ping* messages is
+RECOMMENDED. An unanswered *ping* shall result in a protocol error.
```

Comment:
I'd switch around the order of these two sentences. Additionally, I'd make that protocol error non-binding too, like the rest:

> An unanswered *ping* SHOULD result in a protocol error. A timeout of 30 seconds for unanswered *ping* messages is RECOMMENDED.